From 2c1551695cea056f9eb9e7700930ba8f77238f6d Mon Sep 17 00:00:00 2001
Message-Id: <2c1551695cea056f9eb9e7700930ba8f77238f6d.1434670190.git.chang-joon.lee@intel.com>
In-Reply-To: <5c49cb8f884ecf227e44159a5a4931d2c0814d5c.1434670190.git.chang-joon.lee@intel.com>
References: <5c49cb8f884ecf227e44159a5a4931d2c0814d5c.1434670190.git.chang-joon.lee@intel.com>
From: Xiujun Geng <xiujun.geng@intel.com>
Date: Tue, 12 May 2015 10:46:59 +0800
Subject: [PATCH 3/6] SQUASHME! [VPG]: drm/i915: Use crtc mode to calculate
 scanline

Currently, crtc->hwmode is being used to calculate the scanline
and this is not initialized. Hence the calculated min and max values
during scanline calculation is wrong.
This patch corrects this by using crtc->mode instead
of crtc->hwmode for calculation.

SQUASHME! - This patch should be squashed into the following existing patch:
	Author: Vijay Purushothaman <vijay.a.purushothaman@intel.com>
	Date:   Tue Sep 23 22:43:28 2014 +0530
	FOR_UPSTREAM [VPG]: drm/i915: Start display updates after the vsync
	Change-Id: I48924c71630c4b4b378fd1b41f61ac803161d117

Issue: IMINAN-33414
Change-Id: I8834d0bc1a8a369b4c4bb362e59e107f2f79cc5a
Signed-off-by: Xiujun Geng <xiujun.geng@intel.com>
Signed-off-by: Vandita Kulkarni <vandita.kulkarni@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index a353cea..70a7dbc 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11180,12 +11180,11 @@ int intel_set_disp_commit_regs(struct drm_mode_set_display *disp,
 static unsigned int usecs_to_scanlines(struct drm_crtc *crtc,
 				       unsigned int usecs)
 {
-	/* paranoia */
-	if (!crtc->hwmode.crtc_htotal)
+	if (!crtc->mode.crtc_htotal)
 		return 1;
 
-	return DIV_ROUND_UP(usecs * crtc->hwmode.clock,
-			    1000 * crtc->hwmode.crtc_htotal);
+	return DIV_ROUND_UP(usecs * crtc->mode.clock,
+			    1000 * crtc->mode.crtc_htotal);
 }
 
 static void intel_pipe_vblank_evade(struct drm_crtc *crtc)
@@ -11195,8 +11194,8 @@ static void intel_pipe_vblank_evade(struct drm_crtc *crtc)
 	struct intel_crtc *intel_crtc = to_intel_crtc(crtc);
 	int pipe = intel_crtc->pipe;
 	/* FIXME needs to be calibrated sensibly */
-	u32 min = crtc->hwmode.crtc_vdisplay - usecs_to_scanlines(crtc, 50);
-	u32 max = crtc->hwmode.crtc_vdisplay - 1;
+	u32 min = crtc->mode.crtc_vdisplay - usecs_to_scanlines(crtc, 50);
+	u32 max = crtc->mode.crtc_vdisplay - 1;
 	u32 val;
 
 	local_irq_disable();
-- 
1.9.1

